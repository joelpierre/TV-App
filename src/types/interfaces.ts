export interface ITvSchedule {
  airdate: string;
  airstamp: string;
  airtime: string;
  id: number;
  image?: string;
  name: string;
  number: number;
  runtime: number;
  season: number;
  show: ITvShow;
  summary: string;
  type: string;
  url: string;
}

export interface ITvShow {
  externals: ITvShowExternals;
  genres: string[];
  id: number;
  image: ITvShowImage;
  language: string;
  name: string;
  network: ITvShowNetwork;
  officialSite: string;
  premiered: string;
  rating: ITvShowRating;
  runtime: number;
  schedule: ITvShowSchedule;
  status: string;
  summary: string;
  type: string;
  updated: number;
  url: string;
  webChannel?: string | number | null;
  weight: number;
  _links: ITvShowLinks;
  _embedded: ITvShowEmbedded;
}

export interface ITvShowEmbedded {
  cast: ITvShowCast[];
}

export interface ITvShowCast {
  person: ITvShowPerson;
  character: ITvShowCharacter;
  self: boolean;
  voice: boolean;
}

export interface ITvShowCharacter {
  id: number;
  url: string;
  name: string;
  image?: ITvShowImage;
  _links: ITvShowLinksMeta;
}

export interface ITvShowLinksMeta {
  self: {
    href: string;
  };
}

export interface ITvShowPerson {
  id: number;
  url: string;
  name: string;
  country: ITvShowCountry;
  birthday: string;
  deathday?: string;
  gender: string;
  image: ITvShowImage;
  _links: ITvShowLinksMeta;
}

export interface ITvShowImage {
  medium: string;
  original: string;
}

export interface ITvShowExternals {
  tvrage?: string;
  thetvdb: number;
  imdb: string;
}

export interface ITvShowRating {
  average: number;
}

export interface ITvShowSchedule {
  time: string;
  days: string[];
}

export interface ITvShowCountry {
  code: string;
  name: string;
  timezone: string;
}

export interface ITvShowNetwork {
  id: number;
  name: string;
  country: ITvShowCountry;
}

export interface ITvShowLink {
  href: string;
}

export interface ITvShowLinks {
  nextepisode: ITvShowLink;
  previousepisode: ITvShowLink;
  self: ITvShowLink;
}
