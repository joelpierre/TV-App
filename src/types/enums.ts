/* eslint-disable */
export enum EPageType {
  Home = '/',
  Error404 = 'not-found',
  Show = 'show'
}

export enum EPageHref {
  Home = '/',
  Show = '/show/[slug]'
}
