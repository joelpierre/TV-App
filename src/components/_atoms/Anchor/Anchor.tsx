import React from 'react';
import Link from 'next/link';
import classNames from 'classnames';

import styles from './Anchor.module.scss';
import { EPageHref } from 'common/types/enums';

export interface IAnchorProps {
  className?: string;
  href: EPageHref;
  as: string;
}

const Anchor: React.FunctionComponent<IAnchorProps> = ({
  className,
  href,
  as,
  children
}) => {
  return (
    <Link href={href} as={as} passHref={true}>
      <a className={classNames(styles.Anchor, className)}>
        {children}
      </a>
    </Link>
  );
};

export default Anchor;
