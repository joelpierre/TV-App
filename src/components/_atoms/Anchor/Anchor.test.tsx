import React from 'react';
import { render, RenderResult } from 'test-utils';

import Anchor, { IAnchorProps } from './Anchor';
import { EPageHref } from 'common/types/enums';

const children = 'Test children';
const defaultProps: IAnchorProps = {
  as: '/',
  href: EPageHref.Home
};

const setup = (props?: {}): RenderResult => {
  const setupProps = { ...defaultProps, ...props, children };
  return render(<Anchor {...setupProps} />);
};

describe('<Anchor/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });
  it('renders children', () => {
    expect(wrapper.container.innerHTML).toContain(children);
  });

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
