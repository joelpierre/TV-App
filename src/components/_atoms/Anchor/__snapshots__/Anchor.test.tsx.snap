// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor/> should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="Anchor"
    href="/"
  >
    Test children
  </a>
</DocumentFragment>
`;
